### PR TITLE
JAVA-6350: Add comments for SpringDoc v2 compatible with Spring Boot 3

### PR DESCRIPTION
**Pull request by harry9656**

### CODE REVIEW

File: spring-boot-modules/spring-boot-springdoc/pom.xml [hunk 1]
```diff
@@ -119,53 +118,6 @@
         </resources>
     </build>
 
-    <profiles>
-        <profile>
-            <id>integration</id>
-            <build>
```
Comment (JavierPeris) [inline spring-boot-modules/spring-boot-springdoc/pom.xml:125]: Why has this profile been eliminated?

Comment (harry9656) [reply spring-boot-modules/spring-boot-springdoc/pom.xml:125]: I migrated it to the other module.

Comment (JavierPeris) [reply spring-boot-modules/spring-boot-springdoc/pom.xml:125]: My question is: isn't any other article/code taking advantage of that profile?

Comment (harry9656) [reply spring-boot-modules/spring-boot-springdoc/pom.xml:125]: No, the integration with the plugins is part of this article. Other articles don't refer to this part.

File: spring-boot-modules/spring-boot-springdoc-2/src/main/java/com/baeldung/springdoc/exception/BookNotFoundException.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.baeldung.springdoc.exception;
+
+@SuppressWarnings("serial")
```
[the remark this answers is not in the document]
Comment (harry9656) [reply spring-boot-modules/spring-boot-springdoc-2/src/main/java/com/baeldung/springdoc/exception/BookNotFoundException.java]: We don't need it. I can't find any reason why it was there, so I removed it.

**Pull request merged**